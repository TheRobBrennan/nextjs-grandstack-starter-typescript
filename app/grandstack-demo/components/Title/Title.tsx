import { FC } from "react"

// Material UI
import { Typography } from "@material-ui/core"

interface ITitle {
  children?: string
}

const Title: FC<ITitle> = ({ children }) => {
  return (
    <Typography component="h2" variant="h6" color="primary" gutterBottom>
      {children}
    </Typography>
  )
}
export default Title
