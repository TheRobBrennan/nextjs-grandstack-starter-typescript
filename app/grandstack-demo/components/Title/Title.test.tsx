import React from "react"
import TestRenderer from "react-test-renderer"
import { MockedProvider } from "@apollo/client/testing"

import Title from "./Title"

describe("GRANDstack Title component", () => {
  describe("should render", () => {
    it("without any children", () => {
      const component = TestRenderer.create(
        <MockedProvider mocks={[]} addTypename={false}>
          <Title />
        </MockedProvider>
      )
      expect(component.toJSON()).toMatchSnapshot()
    })
    it("with a supplied title", () => {
      const title = "A GRANDstack Title"
      const component = TestRenderer.create(
        <MockedProvider mocks={[]} addTypename={false}>
          <Title>{title}</Title>
        </MockedProvider>
      )
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
