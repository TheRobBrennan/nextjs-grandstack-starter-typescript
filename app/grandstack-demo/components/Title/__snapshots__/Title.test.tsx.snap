// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GRANDstack Title component should render with a supplied title 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
>
  A GRANDstack Title
</h2>
`;

exports[`GRANDstack Title component should render without any children 1`] = `
<h2
  className="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary MuiTypography-gutterBottom"
/>
`;
